shotover/src/hot_reload/server.rs

```diff
@@ -184,15 +185,47 @@ pub fn start_hot_reload_server(socket_path: String, sources: &[Source]) {
         .collect();
 
     tokio::spawn(async move {
-        match UnixSocketServer::new(socket_path, source_handles) {
-            Ok(mut server) => {
-                info!("Unix socket server started for hot reload communication");
-                if let Err(e) = server.run().await {
-                    error!("Unix socket server error: {:?}", e);
+        const RETRY_DELAY: std::time::Duration = std::time::Duration::from_secs(3);
+        let mut retry_count = 0;
+
+        loop {
+            match UnixSocketServer::new(socket_path.clone(), source_handles.clone()) {
+                Ok(mut server) => {
+                    if retry_count > 0 {
+                        info!(
+                            "Successfully created hot reload server at {} after {} retries",
+                            socket_path, retry_count
+                        );
+                    } else {
+                        info!(
+                            "Hot reload server created at {} on first attempt",
+                            socket_path
+                        );
+                    }
+
+                    // Run the server - this blocks until shutdown
+                    if let Err(e) = server.run().await {
+                        error!("Hot reload server encountered error: {:?}", e);
+                    }
+
+                    info!("Hot reload server at {} has stopped", socket_path);
+                    break;
+                }
+                Err(e) => {
+                    retry_count += 1;
+
+                    // Adaptive logging
+                    if retry_count <= 3 || retry_count % 10 == 0 {
+                        debug!(
```

Comment:
I think this is important enough that it should be `info!` level, so that it actually shows up in standard configurations of shotover.

Everything below `info!` (`debug` and `trace`) are not emitted unless shotover is specifically configured to do so.